druid/src/widget/stepper.rs

```diff
@@ -0,0 +1,208 @@
+                max,
+                min,
+                step,
+                wrap,
+                value_changed: Box::new(value_changed),
+                increase_active: false,
+                decrease_active: false,
+                timer_id: TimerToken::INVALID,
+            },
+        )
+    }
+
+    fn change_value(&mut self, ctx: &mut EventCtx, data: &mut f64, env: &Env) {
+        // increase/decrease value depending on which button is currently active
+        let delta = if self.increase_active {
+            self.step
+        } else if self.decrease_active {
+            -1. * self.step
+        } else {
+            0.0
+        };
+
+        let old_data = *data;
+        *data = (*data + delta).min(self.min).max(self.max);
+
+        if (*data - old_data).abs() > EPSILON {
+            // callback
+            (self.value_changed)(ctx, data, env);
+        } else if self.wrap {
+            if (*data - self.min).abs() < EPSILON {
+                *data = self.max
+            } else {
+                *data = self.min
+            }
+        }
+    }
+}
+
+impl Widget<f64> for Stepper {
+    fn paint(&mut self, paint_ctx: &mut PaintCtx, _data: &f64, env: &Env) {
+        let rounded_rect =
+            RoundedRect::from_origin_size(Point::ORIGIN, paint_ctx.size().to_vec2(), 4.);
+
+        let height = paint_ctx.size().height;
+        let width = env.get(theme::BASIC_WIDGET_HEIGHT);
+        let button_size = Size::new(width, height / 2.);
+
+        paint_ctx.stroke(rounded_rect, &env.get(theme::BORDER), 2.0);
+        paint_ctx.clip(rounded_rect);
+
+        // draw buttons for increase/decrease
+        let increase_button_origin = Point::ORIGIN;
+        let mut decrease_button_origin = Point::ORIGIN;
+        decrease_button_origin.y += height / 2.;
+
+        let increase_button_rect = Rect::from_origin_size(increase_button_origin, button_size);
+        let decrease_button_rect = Rect::from_origin_size(decrease_button_origin, button_size);
+
+        let active_gradient = LinearGradient::new(
+            UnitPoint::TOP,
+            UnitPoint::BOTTOM,
+            (env.get(theme::PRIMARY_LIGHT), env.get(theme::PRIMARY_DARK)),
+        );
+
+        let inactive_gradient = LinearGradient::new(
+            UnitPoint::TOP,
+            UnitPoint::BOTTOM,
+            (env.get(theme::BUTTON_DARK), env.get(theme::BUTTON_LIGHT)),
+        );
+
+        // draw buttons that are currently triggered as active
+        if self.increase_active {
+            paint_ctx.fill(increase_button_rect, &active_gradient);
+        } else {
+            paint_ctx.fill(increase_button_rect, &inactive_gradient);
+        };
+
+        if self.decrease_active {
+            paint_ctx.fill(decrease_button_rect, &active_gradient);
+        } else {
+            paint_ctx.fill(decrease_button_rect, &inactive_gradient);
+        };
+
+        // draw up and down triangles
+        let mut increase_button_arrow = BezPath::new();
```

Comment:
You should be able to have both arrows in one path; after `close_path` you can do `move_path` again.